src/nextStepCliText.ts

```diff
@@ -79,97 +88,145 @@ async function outputLocalSetupCommandsIfProjectsNotAlreadyConfiguredAsLocal(
-    const context = yarnOrNpm === 'both' ? 'yarn and npm' : yarnOrNpm;
-    return `\n\nSetup target projects as local dependencies with ${context} using:`;
+  const argJoin = (arg: string[]) => {
+    return arg.join(' ');
   };
 
-  const commandsToRun = yarnCommandsIfYarnLockPresent.fold(
-    npmCommandsIfPackageLockPresent.map(npmCommandsToRun => {
-      return `${getCommandHeaderString('npm')}\n\t${npmCommandsToRun.join('\n\t')}`;
-    }),
-    yarnCommandsToRun => {
-      return some(
-        npmCommandsIfPackageLockPresent.fold(`${getCommandHeaderString('yarn')}\n\t${yarnCommandsToRun.join('\n\t')}`, npmCommandsToRun => {
-          return `${getCommandHeaderString('both')}\n\t${yarnCommandsToRun.join('\n\t')}\n  and/or\n\t${npmCommandsToRun.join('\n\t')}`;
-        })
-      );
+  const commandForPaths = (command: string, maybePaths: Option<string[]>): string => {
+    return maybePaths.fold('', paths => `\t${command} ${argJoin(paths)}\n`);
+  };
+
+  const thisCommand = cliConstants.commandName;
+  const targetArg = cliConstants.targetProjectArg;
+  const watch = cliConstants.watchCommandArg;
+
+  type StringProducer = () => string;
+  type OutputProvider = string | StringProducer | StringProducer[];
+  type OutputSpecification = OutputProvider | OutputOptions<OutputProvider>;
+
+  const outputSpec: OutputSpecification[] = [
+    `
+
+Set up target projects as local dependencies with `,
+    { npm: 'npm', npmOrYarn: ' or ', yarnClassic: 'yarn', yarnBerry: 'yarn' },
+    ` using:
+`,
+    {
+      npm: [() => commandForPaths('npm install', dependencyPaths), () => commandForPaths('npm install -D', devDependencyPaths)],
+      npmOrYarn: '  and/or\n',
+      yarnClassic: [
+        () => commandForPaths('yarn add', dependencyPaths),
+        () => commandForPaths('yarn add -D', devDependencyPaths),
+        () => '\tyarn install --check-files\n',
+      ],
+      yarnBerry: [
+        () => commandForPaths('yarn add', dependencyReferences),
+        () => commandForPaths('yarn add -D', devDependencyReferences),
+        () => '\tyarn install\n',
+      ],
+    },
+    `
+`,
+    {
+      post: () =>
+        `To get updated changes from target projects, run this command again.
+\t${thisCommand} --${targetArg} ${argJoin(targetProjectPaths)}
+  or watch continually
+\t${thisCommand} ${watch} --${targetArg} ${argJoin(targetProjectPaths)}
+
+`,
+    },
+  ];
+
+  // @ts-ignore noImplicitAny - can't typecheck this recursive return type
+  // Option<string> | (Option<string> | Option<string>[])[] | ...
+  const evaluator = (output?: OutputProvider | OutputOptions<OutputProvider>) => {
+    if (typeof output === 'string') {
+      return some(output);
+    } else if (typeof output === 'function') {
+      return evaluator(output());
+    } else if (Array.isArray(output)) {
+      return output.map(o => evaluator(o));
+    } else if (output && typeof output === 'object') {
+      return Object.keys(output)
+        .sort()
```

Comment:
I like the idea with restructuring this to "get the data" and then "evaluate", but the `Object` support seems fragile. The sorting behavior is what is throwing me the most. The template assumes an order (with `npmOrYarn`) which in theory won't be impacted when the sort occurs but when we need to adjust this the behavior looks like it will break for the unsuspecting use of a key in non-alphabetic order.

Reply:
I think the most important thing is that it work deterministically.  The names of the keys don't end up in the output, so I could rename them `a_npm`, `b_npmOrYarn`, etc. so the dependence on alpha order is more obvious.
(As it turns out, I wanted to prioritize npm before yarn anyway, and alpha order just happened to align with that.)

Reply:
rip out object support, add a conditional function

Reply:
Removed and replaced with boolean && tests